lib/routes/dockerhub/repositories.ts

```diff
@@ -16,27 +15,28 @@ export const route: Route = {
 };
 
 async function handler(ctx: Context) {
-    const owner = ctx.req.param('owner').toLowerCase();
-    const limit = Number.parseInt(ctx.req.query('limit') || '10');
-    const link = `https://hub.docker.com/r/${owner}`;
-    const url = `https://hub.docker.com/v2/repositories/${owner}`;
-    const response = await got(url, {
-        searchParams: {
-            page_size: limit,
-        },
-    });
-    const item = response.data.results.map((repo) => ({
-        title: repo.name,
-        description: `${repo.description}<br>status: ${repo.status_description}<br>stars: ${repo.star_count}<br>pulls: ${repo.pull_count}`,
-        link: `https://hub.docker.com/r/${owner}/${repo.name}`,
-        author: owner,
-        pubDate: parseDate(repo.last_updated),
-        guid: `${owner}/${repo.name}`,
-    }));
+    const { owner, limit } = ctx.req.param();
+    const pageSize = Number.isNaN(Number.parseInt(limit)) ? 10 : Number.parseInt(limit);
+
+    const items = await utils.getRepositories(owner, pageSize);
+    const link = utils.getOwnerLink(owner);
+
     return {
         title: `${owner} repositories`,
         description: `List of repositories for ${owner}`,
         link,
-        item,
-    };
+        language: 'en',
+        item: items.results.map((repo) => {
+            const repositoryLink = utils.getRepositoryLink(owner, repo.name);
+
+            return {
+                title: repo.name,
+                description: `${repo.description}<br>status: ${repo.status_description}<br>stars: ${repo.star_count}<br>pulls: ${repo.pull_count}`,
+                link: repositoryLink,
+                author: owner,
+                pubDate: utils.getPubDate(repo),
+                guid: repositoryLink,
```

Comment:
This changes the guid which causes **all** current subscribers having duplicated entries in their RSS reader.